### PR TITLE
feat(network): add geth connection test

### PR DESCRIPTION
**Pull request by Rjected**
This adds a test that waits for a `SessionEstablished` event from the `Network` after prompting the `GethInstance` to connect to us.

TODO:
 - [ ] longer keepalive tests
 - [x] test outgoing connections
 
~~depends on [ethers-rs#1933](https://github.com/gakonst/ethers-rs/pull/1933), see TODO in the connection test~~
~~depends on #367~~

### CODE REVIEW

File: crates/net/network/tests/it/connect.rs [hunk 1]
```diff
@@ -80,3 +84,52 @@ async fn test_connect_with_boot_nodes() {
...
+    if let Some(ev) = events.next().await {
+        if let NetworkEvent::SessionEstablished { peer_id: incoming_peer_id, .. } = ev {
+            assert_eq!(incoming_peer_id, geth_peer_id);
+        } else {
+            panic!(
+                "unexpected disconnect event, could not establish a session with geth: {:?}",
+                ev
+            );
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Rjected) [reply crates/net/network/tests/it/connect.rs:134]: oh right, that's perfect here

**Review by mattsse — APPROVED** [review]
lgtm,

should we ignore these if geth is not installed?

~~there seems to be an issue with ether's fastrlp license, maybe this is due to an outdated ethers lock?~~

was an issue with Unlicense, fixed here https://github.com/paradigmxyz/reth/pull/366

**Comment by Rjected** [comment]
Yeah, if geth isn't installed then these tests won't work, I'm not sure how best to deal with that

need to fix merge conflicts then this should be good

**Review by mattsse — APPROVED** [review]
lgtm, I think geth install and license error is fixed on main.

can merge after rebase

**Comment by mattsse** [comment]
only 1 smol conflict, rebased.

**Comment by Rjected** [comment]
test errors unrelated, merging

**Pull request merged**